src/main/java/seedu/address/model/module/TutorialTeam.java

```diff
@@ -0,0 +1,190 @@
+package seedu.address.model.module;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.ArrayList;
+
+import seedu.address.model.person.Person;
+
+/**
+ * Represents a Module's tutorial team.
+ * Guarantees: immutable; is valid as declared in
+ * {@link #isValidTeamName(String)}
+ */
+public class TutorialTeam {
+
+    public static final String MESSAGE_CONSTRAINTS = "Please enter a valid tutorial team name "
+            + "eg. Team 1, and it should not be blank";
+    public static final String MESSAGE_NAME_CONSTRAINTS = "Team names should only contain alphanumeric "
+            + "characters and spaces, and it should not be blank";
+    public static final String MESSAGE_SIZE_CONSTRAINTS = "Team size should be a positive integer";
+
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String teamName;
+    public final int teamSize;
+    private final ArrayList<Person> students;
+
+    /**
+     * Constructs a {@code TutorialTeam} with default values.
+     * Initializes the {@code value} field to an empty string, creates an empty
+     * list for {@code students} and sets the {@code teamSize} to the maximum.
+     */
+    public TutorialTeam() {
+        this.teamName = "";
+        this.students = new ArrayList<>();
+        this.teamSize = Integer.MAX_VALUE;
+    }
+
+    /**
+     * A constructor for TutorialTeam. Creates a tutorial team of a certain size
+     * with no students.
+     * @param tutorialTeam
+     * @param teamSize
+     */
+    public TutorialTeam(String tutorialTeam, int teamSize) {
+        requireAllNonNull(tutorialTeam, teamSize);
+        checkArgument(isValidTeamName(tutorialTeam), MESSAGE_NAME_CONSTRAINTS);
+        checkArgument(isValidSize(teamSize), MESSAGE_SIZE_CONSTRAINTS);
+        this.teamName = tutorialTeam;
+        this.students = new ArrayList<>();
+        this.teamSize = teamSize;
+    }
+
+    /**
+     * A constructor for TutorialTeam. Creates a tutorial team with students.
+     * @param tutorialClass to be added
+     * @param students      in the tutorial class
+     */
+    public TutorialTeam(String tutorialTeam, ArrayList<Person> students) {
+        requireAllNonNull(tutorialTeam);
+        checkArgument(isValidTeamName(tutorialTeam), MESSAGE_NAME_CONSTRAINTS);
+        this.teamName = tutorialTeam;
+        this.students = students;
+        this.teamSize = Integer.MAX_VALUE;
+    }
```

Comment:
If we set teamSize = Integer.MAX_VALUE, would there be a possibility that we can add that many number of students/or an extremely large number of students to the team which may cause our application to be extremely slow?

Reply:
To understand the rationale behind Integer.MAX_VALUE, there are 2 user flows.

1. Team size defined -> teamSize set to defined int value.

2. Team size undefined -> By default, we can set to null for teamSize but it is better to assign an integer value (in this case Integer.MAX_VALUE) so that we can handle the variable type easily and ensure that it is always int.

Hope this answers your question :)